diff --git a/src/melee/lb/lb_013B.c b/src/melee/lb/lb_013B.c
index 75dfb9cef..6507cc215 100644
--- a/src/melee/lb/lb_013B.c
+++ b/src/melee/lb/lb_013B.c
@@ -22,7 +22,7 @@ bool lb_80013BB0(ColorOverlay* arg)
 
 bool lb_80013BB8(ColorOverlay* arg0)
 {
-    arg0->x0_timer += arg0->x8_ptr1->unk.timer;
+    arg0->x0_timer += CO_X8(arg0->x8_ptr1)->unk.timer;
     ++arg0->x8_ptr1;
     return false;
 }
@@ -36,9 +36,9 @@ bool lb_80013BE4(ColorOverlay* arg0)
 
 static bool lb_80013C18(ColorOverlay* arg0)
 {
-    arg0->x7C_light_enable = arg0->x8_ptr1->light_rot2.light_enable;
-    arg0->x74_light_rot_x = arg0->x8_ptr1->light_rot2.x;
-    arg0->x78_light_rot_yz = arg0->x8_ptr1->light_rot2.yz;
+    arg0->x7C_light_enable = CO_X8(arg0->x8_ptr1)->light_rot2.light_enable;
+    arg0->x74_light_rot_x = CO_X8(arg0->x8_ptr1)->light_rot2.x;
+    arg0->x78_light_rot_yz = CO_X8(arg0->x8_ptr1)->light_rot2.yz;
     ++arg0->x8_ptr1;
     arg0->x50_light_color.r = arg0->x8_ptr1->light_color.r;
     arg0->x50_light_color.g = arg0->x8_ptr1->light_color.g;
@@ -78,7 +78,7 @@ static bool lb_80013D68(ColorOverlay* arg0)
 
 static bool lb_80013E3C(ColorOverlay* arg0)
 {
-    float f = arg0->x8_ptr1++->unk.timer;
+    float f = CO_X8(arg0->x8_ptr1++)->unk.timer;
     arg0->x64_lightblend_red =
         ((0.5f + arg0->x8_ptr1->light_color.r) - arg0->x50_light_color.r) / f;
     arg0->x68_lightblend_green =
@@ -93,8 +93,8 @@ static bool lb_80013E3C(ColorOverlay* arg0)
 
 static bool lb_80013F78(ColorOverlay* arg0)
 {
-    arg0->x74_light_rot_x = arg0->x8_ptr1->light_rot1.x;
-    arg0->x78_light_rot_yz = arg0->x8_ptr1->light_rot1.yz;
+    arg0->x74_light_rot_x = CO_X8(arg0->x8_ptr1)->light_rot1.x;
+    arg0->x78_light_rot_yz = CO_X8(arg0->x8_ptr1)->light_rot1.yz;
     ++arg0->x8_ptr1;
     return false;
 }
@@ -128,7 +128,7 @@ static bool lb_80014014(ColorOverlay* arg0)
 
 static bool lb_800140F8(ColorOverlay* arg0)
 {
-    float f = arg0->x8_ptr1++->unk.timer;
+    float f = CO_X8(arg0->x8_ptr1++)->unk.timer;
     arg0->x40_colorblend_red =
         ((0.5f + arg0->x8_ptr1->light_color.r) - arg0->x2C_hex.r) / f;
     arg0->x44_colorblend_green =
@@ -168,7 +168,7 @@ bool lb_80014258(Fighter_GObj* gobj, void* arg1, FtCmd2 cmd)
     }
 
     while (co->x8_ptr1 != NULL && co->x0_timer == 0) {
-        u32 opcode = co->x8_ptr1->unk.unk;
+        u32 opcode = CO_X8(co->x8_ptr1)->unk.unk;
         if (!Command_Execute((CommandInfo*) co, opcode)) {
             if (opcode < 0x15U) {
                 u32 idx = opcode - 0xA;
