diff --git a/src/melee/lb/lbcommand.c b/src/melee/lb/lbcommand.c
index 5ea9c63cc..91b316dc0 100644
--- a/src/melee/lb/lbcommand.c
+++ b/src/melee/lb/lbcommand.c
@@ -19,14 +19,14 @@ void Command_00(CommandInfo* info)
 /// SynchronousTimer
 void Command_01(CommandInfo* info)
 {
-    info->timer += info->u->Command_00.value;
+    info->timer += CMD_U(info)->Command_00.value;
     NEXT_CMD(info);
 }
 
 /// AsynchronousTimer
 void Command_02(CommandInfo* info)
 {
-    info->timer = info->u->Command_02.value - info->frame_count;
+    info->timer = CMD_U(info)->Command_02.value - info->frame_count;
     NEXT_CMD(info);
 }
 
@@ -35,7 +35,7 @@ void Command_03(CommandInfo* info)
 {
     info->event_return[info->loop_count++] = info->u + 1;
     info->event_return[info->loop_count++] =
-        (union CmdUnion*) info->u->Command_03.value;
+        (union CmdUnion*) CMD_U(info)->Command_03.value;
     NEXT_CMD(info);
 }
 
@@ -83,8 +83,8 @@ void Command_08(CommandInfo* info)
 
 void Command_09(CommandInfo* info)
 {
-    lbBgFlash_80021C48(info->u->Command_09.param_1,
-                       info->u->Command_09.param_2);
+    lbBgFlash_80021C48(CMD_U(info)->Command_09.param_1,
+                       CMD_U(info)->Command_09.param_2);
     NEXT_CMD(info);
 }
 
